src/saving/InProgressLoad.cs

```diff
@@ -88,10 +88,14 @@ private void Step()
                 break;
             case State.ReadingData:
             {
+                // Wait for transition to finish.
```

Comment:
This class adds the transition on line 82, so shouldn't it be possible for this to also actually register the proper callback? I think checking if there are queued transitions is a dirty way to do it and I think we've managed to avoid using that elsewhere in the codebase.

Reply:
Sure.